fixes #17510 - specify which columns to export

thomasmckay opened the pull request:
Added --export-columns to limit which columns are exported
```
$hammer csv content-hosts --export --export-columns "Name,Organization,Content View"
Name,Organization,Content View
guest0,Example Corporation,Default Organization View
guest1,Example Corporation,Default Organization View
host0,Example Corporation,Default Organization View
katello.example.com,Example Corporation,Default Organization View
```

Also added a --help feature to list columns
```
$ hammer csv content-hosts --help
Usage:
    hammer csv content-hosts [OPTIONS]

Options:
 --continue-on-error           Continue processing even if individual resource error
 --export                      Export current data instead of importing
 --export-columns COLUMN_NAMES Comma separated list of column names to export
 --file FILE_NAME              CSV file (default to /dev/stdout with --export, otherwise required)
 --itemized-subscriptions      Export one subscription per row, only process update subscriptions on import
 --organization ORGANIZATION   Only process organization matching this name
 -h, --help                    print help
 -v, --verbose                 be verbose

Columns:
 Name - Name of resource
 Search - Search for matching names during import (overrides 'Name' column)
 Organization - Organization name
 Environment - Lifecycle environment name
 Content View - Content view name
 Host Collections - Comma separated list of host collection names
 Name - Is a virtual host, Yes or No
 Host - Hypervisor host name for virtual hosts
 OS - Operating system
 Arch - Architecture
 Sockets - Number of sockets
 RAM - Quantity of RAM in bytes
 Cores - Number of cores
 SLA - Service Level Agreement value
 Products - Comma separated list of products, where product is of format "<sku>|<name>"
```

In cli_config.yml
```
:csv:
  :enable_module: true
  :columns:
    :content-hosts:
      :define:
        -
          :name: Subscription Status
          :json:
            - subscription_status_label
        -
          :name: Last Checkin
          :json:
            - subscription_facet_attributes
            - last_checkin
      :export:
        - Name
        - Organization
        - Subscription Status
        - Last Checkin
```

and then

```
$ hammer csv content-hosts --export
Name,Organization,Subscription Status,Last Checkin
guest0,Example Corporation,Unentitled,2016-11-22 14:43:06 UTC
guest1,Example Corporation,Unentitled,2016-11-22 14:47:25 UTC
host0,Example Corporation,Unentitled,2016-11-22 14:06:35 UTC
katello.example.com,Example Corporation,Partially entitled,2016-11-22 23:40:04 UTC
```
or
```
$ hammer csv content-hosts --export --export-columns "Name,Subscription Status"
Name,Subscription Status
guest0,Unentitled
guest1,Unentitled
host0,Unentitled
katello.example.com,Partially entitled
```

thomasmckay commented:
@akofink - what do you think?

akofink commented:
I really like the idea, but is this only for export? The exported CSV would cause issues with import, considering some information is required to reconstruct most models.

thomasmckay commented:
Yes, only for export. I am slowly trying to build in nice handling for missing columns since most of the code relies on their presence and leaving one out could potentially clear the value unintentionally. Importing custom columns may work in some cases but much of katello/foreman does not handle the "PUT of a GET".

thomasmckay commented on lib/hammer_cli_csv/content_hosts.rb:
```diff
@@ -12,6 +12,7 @@ def self.supported?
       end
 
       option %w(--itemized-subscriptions), :flag, _('Export one subscription per row, only process update subscriptions on import')
+      option %w(--export-columns), 'COLUMN_NAMES', _('Comma separated list of column names to export')
```
I'll change this option from _--export-columns_ to simplye _--columns_ to allow the import of only specific columns.

thomasmckay commented on lib/hammer_cli_csv/content_hosts.rb:
```diff
@@ -28,6 +29,58 @@ def self.supported?
+         _(" %{name} - Number of cores") % {:name => CORES},
+         _(" %{name} - Service Level Agreement value") % {:name => SLA},
+         _(" %{name} - Comma separated list of products, where product is of format \"<sku>|<name>\"") % {:name => PRODUCTS},
+        ].join("\n")
```
Include the custom columns. Add a _help_ field to the yaml.

thomasmckay commented:
@akofink - This is ready for some poking and prodding.

I changed the CLI option too --columns, though I left the description as export. Added an issue for using it on import http://projects.theforeman.org/issues/17511

thomasmckay commented:
@akofink - Interesting travis failures... can't read the config.yml. Any suggestions? I'm surprised that coveralls went down since I wrote a bunch of tests for content-hosts; could just be the failed config.
